pkg/apis/config/defaults_test.go

```diff
@@ -209,12 +226,23 @@ func TestTemplating(t *testing.T) {
 		})
 	}
 	t.Run("bad-template", func(t *testing.T) {
-		if _, err := NewDefaultsConfigFromConfigMap(&corev1.ConfigMap{
+		defCM, err := NewDefaultsConfigFromConfigMap(&corev1.ConfigMap{
 			Data: map[string]string{
 				"container-name-template": "{{animals-being-bros]]",
-			},
-		}); err == nil {
+			}})
+		if err == nil {
 			t.Error("Expected an error but got none.")
 		}
+		def, err := NewDefaultsConfigFromMap(
+			map[string]string{
+				"container-name-template": "{{animals-being-bros]]",
+			})
+		if err == nil {
+			t.Error("Expected an error but got none.")
+		}
+
+		if diff := cmp.Diff(defCM, def); diff != "" {
+			t.Errorf("Config mismatch: diff(-want,+got):\n%s", diff)
+		}
```

Comment:
Why this check if we always expect error, this should be nil?

Reply:
good point, removed the diff check